arex-storage-web-api/src/main/java/com/arextest/storage/service/DatabaseParseService.java

```diff
@@ -133,31 +147,48 @@ public static List<String> parseTableNames(String operationName) {
      * @param sql sql
      * @return table schema info
      */
-    @VisibleForTesting
-    public static TableSchema parse(String sql) {
+    public TableSchema parse(String sql, String appId) {
+        long startTime = System.currentTimeMillis();
+
         TableSchema tableSchema = new TableSchema();
         try {
             sql = PATTERN.matcher(sql).replaceAll(" ");
 
             Statement statement = CCJSqlParserUtil.parse(sql);
             tableSchema.setAction(getAction(statement));
 
-            List<String> tableNameList = new TablesNamesFinder().getTableList(statement);
-            // sort table name
+            Set<String> tableNameList = new TablesNamesFinder().getTables(statement);
+            // sort table nam, Sort in lexicographical order
             if (CollectionUtils.isNotEmpty(tableNameList)) {
-                Collections.sort(tableNameList);
+                tableNameList = new TreeSet<>(tableNameList);
             }
             tableSchema.setTableNames(tableNameList);
         } catch (Throwable e) {
-            LOGGER.warn("parse sql error, sql: {}", sql, e);
+            return tableSchema;
```

Comment:
I have a question. When catching Throwable exceptions and oom occurs, will the log still be printed?

Reply:
Won't